src/main/java/de/fraunhofer/aisec/cpg/frontends/cpp/ExpressionHandler.java

```diff
@@ -394,12 +391,9 @@ private CallExpression handleFunctionCallExpression(CPPASTFunctionCallExpression
     if (reference instanceof MemberExpression) {
       String baseTypename;
       // Pointer types contain * or []. We do not want that here.
-      Type baseType = ((Expression) ((MemberExpression) reference).getBase()).getType();
-      if (baseType instanceof PointerType) {
-        baseTypename = (((PointerType) baseType).getElementType().getTypeName());
-      } else {
-        baseTypename = baseType.getTypeName();
-      }
+      Type baseType = ((Expression) ((MemberExpression) reference).getBase()).getType().getRoot();
+      assert !(baseType instanceof SecondOrderType);
```

Comment:
Do we have a policy about the usage of `assert`? Maybe through our custom `TranslationException` instead?

Reply:
AFAIK there is no poliy. But for me, it was not obvious that `getRoot()` is supposed to guarantee that the result is _not_ a `PointerType` and _not_ a `ReferenceType`. This is not clearly documented and it does not become obvious from the method's name so I made it explicit with an `assert` for my future self to remember.
`asserts` should not add overhead to runtime, so it's a matter of taste.